lib/OpenQA/Worker/WebUIConnection.pm

```diff
@@ -83,6 +83,8 @@ sub register ($self) {
     my $ua = $self->ua;
     my $url = $self->url->clone;
     die 'client not correctly initialized before registration' unless ($webui_host && $working_dir && $ua && $url);
+    die "Rejecting registration as ambiguous 'localhost' to remote webui_host '$webui_host'"
+      if $worker->worker_hostname eq 'localhost' && $webui_host !~ qr/localhost/;
```

Comment:
This breaks if one uses e.g. 127.0.0.1. We already have a utility to cover this somewhere so please use it here.

Reply:
Don't just die here; this will not lead to the desired error handling. The line above is ok because it acts as an internal assert (and if it would ever fire we have a programming mistake to fix). However, your line is supposed to be logged as a normal error message and the worker is supposed to keep running (as there might be other web UI connections that are not broken). So do the error handling like in the `!defined $worker_id` case below.